container/docker/engine.py

```diff
@@ -213,6 +213,15 @@ def run_conductor(self, command, config, base_path, params, engine_name=None, vo
 
         if not volumes:
             volumes = {}
+
+        if params.get('with_volumes'):
+          for volume in params.get('with_volumes'):
+              volume_parts = volume.split(':')
+              volumes[volume_parts[0]] = {
+                  'bind': volume_parts[1],
+                  'mode': volume_parts[2] if len(volume_parts) > 2 else 'rw'
+              }
+
```

Comment:
Good catch. I was actually looking at the docker cli docs and could not find explicit support for this. I'll update the PR